### PR TITLE
Group 2: Passing tests

### PR DESCRIPTION
**Pull request by lalvarezguillen**
G2: David and Luis

### CODE REVIEW

File: simple_database/dbmodule.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        for row in self.rows:
+            if list(kwarg.values())[0] in getattr(row, list(kwarg.keys())[0]):
+                results.append(row)
+        return iter(results)        
+    
+    def insert(self, *args): 
+        if len(args) != len(self.columns):
+            raise ValidationError("Invalid amount of field")
+        for index, arg in enumerate(args):
+            if type(arg).__name__ != self.columns[index]['type']:
```
Comment (jon003) [inline simple_database/dbmodule.py:76]: I wish I had thought to look at **name** !

File: simple_database/dbmodule.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                raise ValidationError(
+                    'Invalid type of field "{}": Given "{}", expected "{}"'.format(
+                        self.columns[index]["name"], type(arg).__name__, self.columns[index]['type']
+                    )
+                )
+        self.rows.append(Row(self.columns, *args))
+        self.db.commit() #After an insertion, update the DB
+    
+    
+class Row(object):
```
Comment (jon003) [inline simple_database/dbmodule.py:86]: We had a similar row object.

File: simple_database/dbmodule.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import json, os, pickle
```
Comment (jon003) [inline simple_database/dbmodule.py:1]: Why both json and pickle?  Seems a little bit superfluous.

File: simple_database/dbmodule.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            if temp_file: temp_file = json.loads(temp_file)
+            else: temp_file = {}
+            temp_file[db_name] = db.path
+        with open(os.path.join(BASE_DB_FILE_PATH, "path_index.json"), "w") as file:
+            file.write(json.dumps(temp_file))
+        return db
+        
+    # raise NotImplementedError()
+
+def connect_database(db_name):
```
Comment (jon003) [inline simple_database/dbmodule.py:33]: Your connect and create are very close.  Could they have been merged into one function, with perhaps a default parameter over-ridden for a connect?

File: simple_database/dbmodule.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                db = pickle.load(pickled_file)
+                return db
+ 
+        
+    #error if doesnt exist
+
+class Table(object):
+    def __init__(self, db, name, columns):
+        self.db = db
+        self.name = name
```
Comment (jon003) [inline simple_database/dbmodule.py:50]: You never end up using self.name that I can see.

File: simple_database/dbmodule.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    
+    
+class Row(object):
+    def __init__(self, structure, *args):     #Row(columns, [1,2,3,4])
+        for index, arg in enumerate(args):
+            setattr(self, structure[index]["name"], arg)
+            
+class Db(object):
+    def __init__(self, name):
+        self.name = name
```
Comment (jon003) [inline simple_database/dbmodule.py:93]: Not seeing self.name used here either.

File: simple_database/main.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
-def create_database(db_name):
-    raise NotImplementedError()
+# def create_database(db_name):
```
Comment (jon003) [inline simple_database/main.py:1]: Why not eliminate the effectively empty file?  or move your sub file into this one, if you are exposing your user functions via dbmodule.py?

File: tests/test_main.py [hunk 8]
```diff
@@ -5,7 +5,7 @@
 import unittest
 from datetime import date
 
-from simple_database import create_database, connect_database
+from simple_database.dbmodule import create_database, connect_database
```
Comment (jon003) [inline tests/test_main.py:8]: if you moved your imports into the main.py, you would not have to alter the tests.

File: simple_database/dbmodule.py (hunk 1 above)
Comment (lukepensiero) [inline simple_database/dbmodule.py:76]: Wow, that's great. After we discussed how everything in Python is an object, too. We never thought to look for the name of the 'int' object.

**Pull request closed without merging**